internal/processing/authorization_policy_helpers.go

```diff
@@ -0,0 +1,45 @@
+package processing
+
+import (
+	"fmt"
+	"istio.io/api/security/v1beta1"
+
+	gatewayv1beta1 "github.com/kyma-incubator/api-gateway/api/v1beta1"
+	"github.com/kyma-incubator/api-gateway/internal/builders"
+	securityv1beta1 "istio.io/client-go/pkg/apis/security/v1beta1"
+)
+
+func modifyAuthorizationPolicy(existing, required *securityv1beta1.AuthorizationPolicy) {
+	existing.Spec = *required.Spec.DeepCopy()
+}
+
+func generateAuthorizationPolicy(api *gatewayv1beta1.APIRule, rule gatewayv1beta1.Rule, additionalLabels map[string]string) *securityv1beta1.AuthorizationPolicy {
+	namePrefix := fmt.Sprintf("%s-", api.ObjectMeta.Name)
+	namespace := api.ObjectMeta.Namespace
+	ownerRef := generateOwnerRef(api)
+
+	arBuilder := builders.AuthorizationPolicyBuilder().
+		GenerateName(namePrefix).
+		Namespace(namespace).
+		Owner(builders.OwnerReference().From(&ownerRef)).
+		Spec(builders.AuthorizationPolicySpecBuilder().From(generateAuthorizationPolicySpec(rule))).
+		Label(OwnerLabel, fmt.Sprintf("%s.%s", api.ObjectMeta.Name, api.ObjectMeta.Namespace)).
+		Label(OwnerLabelv1alpha1, fmt.Sprintf("%s.%s", api.ObjectMeta.Name, api.ObjectMeta.Namespace))
+
+	for k, v := range additionalLabels {
+		arBuilder.Label(k, v)
+	}
+
+	return arBuilder.Get()
+}
+
+func generateAuthorizationPolicySpec(rule gatewayv1beta1.Rule) *v1beta1.AuthorizationPolicy {
+	authorizationPolicySpec := builders.AuthorizationPolicySpecBuilder().
+		Selector(builders.SelectorBuilder().MatchLabels("app", *rule.Service.Name)).
```

Comment:
we want to support matching service by `Namespace` as well, which is currently not quite straight forward with https://istio.io/latest/docs/reference/config/type/workload-selector/
another way ? or at least address it and tackle it later.

Reply:
what about adding the AP's and RA's in same namespace as the service ? If I understand this correctly from the documentation: `The scope of label search is restricted to the configuration namespace in which the resource is present.`